src/app/components/template-button/template-button.component.ts

```diff
@@ -27,6 +29,30 @@ export class TemplateButtonComponent {
     }
 
     processMouseClick(e: MouseEvent): void {
-        console.log(`Template button "${this.buttonText}" clicked`, e);
+        if (this.buttonAction) {
+            this.buttonAction.emit();
+        }
+    }
+
+    processDrop(e: DragEvent): void {
+        this.prevent(e);
+        const target = e.target as HTMLElement;
+        target.classList.remove('drag-hover');
+
+        if (this.dropAction) {
+            this.dropAction.emit(e);
+        }
```

Comment:
Das wird ja von der Klasse selbst gesetzt, das kann effektiv nicht null sein.
Bei `@Input` Attributen kann das sein, weil die von außen belegt werden. Wenn `@Outputs` nicht belegt sind, gehen die emittierten Werte einfach ins leere

```suggestion
        this.dropAction.emit(e);
```